src/DynamicRupture/Output/Builders/ReceiverBasedOutputBuilder.cpp

```diff
@@ -0,0 +1,190 @@
+    }
+    {
+      auto faceAlignedToGlb =
+          init::T::view::create(outputData->faceAlignedToGlbData[receiverId].data());
+      auto glbToFaceAligned =
+          init::Tinv::view::create(outputData->glbToFaceAlignedData[receiverId].data());
+
+      seissol::model::getFaceRotationMatrix(
+          faceNormal, tangent1, tangent2, faceAlignedToGlb, glbToFaceAligned);
+    }
+  }
+}
+
+void ReceiverBasedOutputBuilder::initOutputVariables(
+    std::array<bool, std::tuple_size<DrVarsT>::value>& outputMask) {
+  auto assignMask = [&outputMask](auto& var, int receiverId) {
+    var.isActive = outputMask[receiverId];
+  };
+  misc::forEach(outputData->vars, assignMask);
+
+  auto allocateVariables = [this](auto& var, int) {
+    var.maxCacheLevel = outputData->maxCacheLevel;
+    var.allocateData(this->outputData->receiverPoints.size());
+  };
+  misc::forEach(outputData->vars, allocateVariables);
+}
+
+void ReceiverBasedOutputBuilder::initJacobian2dMatrices() {
+  const auto& faultInfo = meshReader->getFault();
+  const auto& verticesInfo = meshReader->getVertices();
+  const auto& elementsInfo = meshReader->getElements();
+
+  size_t nReceiverPoints = outputData->receiverPoints.size();
+  outputData->jacobianT2d.resize(nReceiverPoints);
+
+  for (size_t receiverId = 0; receiverId < nReceiverPoints; ++receiverId) {
+    auto side = outputData->receiverPoints[receiverId].localFaceSideId;
+    auto elementIndex = outputData->receiverPoints[receiverId].elementIndex;
+
+    assert(elementIndex >= 0);
+
+    const auto& element = elementsInfo[elementIndex];
+    auto face = getGlobalTriangle(side, element, verticesInfo);
+
+    VrtxCoords xab, xac;
+    {
+      constexpr size_t x{0}, y{1}, z{2};
+      xab[x] = face.p2[x] - face.p1[x];
+      xab[y] = face.p2[y] - face.p1[y];
+      xab[z] = face.p2[z] - face.p1[z];
+
+      xac[x] = face.p3[x] - face.p1[x];
+      xac[y] = face.p3[y] - face.p1[y];
+      xac[z] = face.p3[z] - face.p1[z];
+    }
+
+    auto faultIndex = outputData->receiverPoints[receiverId].faultFaceIndex;
+    const auto* tangent1 = faultInfo[faultIndex].tangent1;
+    const auto* tangent2 = faultInfo[faultIndex].tangent2;
+
+    Eigen::Matrix<real, 2, 2> matrix;
+    matrix(0, 0) = MeshTools::dot(tangent1, xab);
+    matrix(0, 1) = MeshTools::dot(tangent2, xab);
+    matrix(1, 0) = MeshTools::dot(tangent1, xac);
+    matrix(1, 1) = MeshTools::dot(tangent2, xac);
+    outputData->jacobianT2d[receiverId] = matrix.inverse();
+  }
+}
+
+void ReceiverBasedOutputBuilder::assignNearestInternalGaussianPoints() {
+  auto& geoPoints = outputData->receiverPoints;
+  constexpr int numPoly = CONVERGENCE_ORDER - 1;
+
+#ifndef stroud
+  const int rank = seissol::MPI::mpi.rank();
+  logWarning(rank)
+      << "This quadrature rule does not move gaussian points away from cell edges when"
+      << "evaluating rupture speed. This may result in inacurrate rupture speeds there.";
```

Comment:
to evaluate rupture speed you need to compute the derivative of the rupture time.
We do not evaluate it at the (sub)triangle center but at the nearest quadrature point.
With the old quadrature rule, this was relatively inacurate for the most near-edge quadrature points (actually probably on the sides of the triangle not well sampled by the uneven quadrature rule).
Therefore we were assessing rupture speed at a quadrature points more inside the element at these quadrature points.
This is not needed with the new quadrature rule, because it covers much better the triangle.

Anyway, I would suggest removing altogether the warning